### ISSUE TITLE
[windows] AltGr-characters collide with Ctrl+Alt hotkeys

### ISSUE THREAD

**Issue by yalov** [atom/atom-keymap#211]
There is a thing like “the 3d and the 4th layers of keyboard layout” (`AltGr`, `AltGr`+`Shift`), it is not few-men keyboard layout — many standard native keyboard layout has that 3 layer (`AltGr`). 
I think all that ö, ü, ć, ź, ĉ, ẑ, etc, is placed on 3 layer. It collide with `Ctrl`+`Alt` hotkeys.

Even more, you can create [your own](https://yalov.github.io/images/yarbur.png) layout through “Microsoft Keyboard Layout Creator” or use [some](http://ilyabirman.net/projects/typography-layout/) popular “typography layout”, so every key on keyboard can have that 3 layer, and user can want to use or 3 layer or hotkeys.

For example, if you do some C++ coding, then you are usually stick to ACSII (beside of commenting), and ö is not really matter, but if you write some .md file — you need all your « „ “ × — ≠ ± ∞ ĉ ♪ © € ¢ ₽  ₴».

Public custom keystroke resolver API in `init.coffee` for every key it is cool, but for some user there are so many keys, that **main setting switch (`Ctrl`+`Alt` hotkeys <--> `AltGr` 3-4 layers) is better**. It is not replace, it is addition.

P.S. there are 3d layer — `AltGr`, 4th layer — `AltGr`+`Shift`, and `AltGr` = `Ctrl`+`Alt`, so the problem area is
`Ctrl`+`Alt`,  `Ctrl`+`Alt`+`Shift`, `AltGr`, `AltGr`+`Shift`

P.S.2 I think switch should work only if there are colliding. 
For example, if there are `Ctrl`+`Alt`+`b` hotkey for "beautify", `AltGr`+`n` for "ň", and no  `AltGr`+`b` character, no `Ctrl`+`Alt`+`n` hotkey, then they should work both, switch-independently.

**Comment by yalov** [comment]
@Ben3eeE, @neochief

**Comment by Ben3eeE** [comment]
@yalov @neochief I created a package that uses the [custom keystroke resolver API](https://atom.io/docs/api/v1.14.4/KeymapManager#instance-addKeystrokeResolver) to always honor the AltGraph modified character if one exists. Check it out here https://atom.io/packages/honor-altgraph

I gave it a quick test using the typography layout that you linked but I am not very familiar with this layout. If you run into any issues let me know by opening a new issue on the package repository.

Thanks for opening this issue 🙇 we might add this as a configuration option in Atom later but for now I will close this as resolved.
